deepchem/models/tests/test_layers.py

```diff
@@ -1450,3 +1450,83 @@ def test_torch_highway_layer():
     output_tensor = highway_layer(input_tensor)
 
     assert output_tensor.shape == (batch_size, feat_dim)
+
+
+@pytest.mark.torch
+def test_torch_attn_lstm_embedding():
+    """Test invoking AttnLSTMEmbedding."""
+    max_depth = 1
+    n_test = 2
+    n_support = 3
+    n_feat = 3
+    layer = torch_layers.AttnLSTMEmbedding(n_test, n_support, n_feat, max_depth)
+    layer.lstm.W = torch.tensor(
+        [[
+            -0.1273, -0.1250, 0.3918, 0.2397, -0.4295, 0.4657, -0.3275, -0.4658,
+            -0.1461, 0.4497, 0.3217, 0.4292
+        ],
+         [
+             0.2073, 0.0721, 0.1439, 0.4549, -0.5348, -0.2929, 0.1359, 0.0471,
+             0.2583, 0.2730, -0.1592, -0.3192
+         ],
+         [
+             -0.0468, 0.5013, -0.2015, -0.4979, -0.3264, -0.5235, -0.3467,
+             0.2771, -0.0445, 0.4920, -0.5284, -0.1588
+         ],
+         [
+             -0.4745, 0.3819, 0.0476, -0.2088, 0.5514, 0.4658, -0.3047, -0.4388,
+             -0.3003, 0.4436, 0.5597, 0.0759
+         ],
+         [
+             0.1443, 0.1653, 0.2448, -0.2557, -0.3888, 0.3011, -0.4685, -0.2519,
+             0.1655, -0.3773, -0.4264, 0.3108
+         ],
+         [
+             0.0050, -0.1415, -0.5631, 0.3684, 0.2077, -0.1922, 0.3428, 0.5133,
+             -0.3577, 0.1347, 0.2427, 0.0068
+         ]],
+        dtype=torch.float32)
+    layer.lstm.U = torch.tensor(
+        [[
+            0.0990, 0.0013, -0.5093, -0.1915, 0.1896, -0.0151, 0.1051, -0.1788,
+            0.7236, -0.2065, 0.0213, -0.2197
+        ],
+         [
+             -0.2521, -0.1700, 0.2047, -0.2722, -0.0880, 0.3533, 0.3856, 0.3532,
+             -0.0856, -0.3557, 0.0102, -0.5017
+         ],
+         [
+             0.1716, 0.3322, -0.0321, -0.0234, 0.2916, -0.0215, 0.0500, 0.2335,
+             -0.1415, -0.5520, 0.5093, 0.3633
+         ]],
+        dtype=torch.float32)
+    layer.lstm.b = torch.tensor(
+        [0., 0., 0., 1., 1., 1., 0., 0., 0., 0., 0., 0.], dtype=torch.float32)
+
+    test = np.array([[0.06616174, 0.8224941, 0.22045802],
+                     [0.43049398, 0.9574964, 0.62803596]],
+                    dtype=np.float32)
+    support = np.array([[0.52633643, 0.727206, 0.44552732],
+                        [0.13802947, 0.5601545, 0.62722725],
+                        [0.23823564, 0.40358722, 0.7951269]],
+                       dtype=np.float32)
+    test_out_tf = torch.tensor(
+        [[0.0292, 0.8319, 0.1675], [0.3945, 0.9680, 0.5746]],
+        dtype=torch.float32)
+
+    test_out, support_out = layer([test, support])
+
+    assert test_out.shape == (n_test, n_feat)
+    assert support_out.shape == (n_support, n_feat)
+    assert torch.allclose(test_out_tf, test_out, atol=1e-04)
+    assert torch.allclose(torch.tensor(support), support_out, atol=1e-04)
+
+
+@pytest.mark.torch
+def test_torch_cosine_dist():
```

Comment:
@Dragonwagon18 Let's add in a test with an array of just zeros to ensure if the epsilon term is working